actors/builtin/market/market_actor.go

```diff
@@ -556,22 +532,57 @@ func (a Actor) CronTick(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
 	return nil
 }
 
+//
+// Exported functions
+//
+
+// Validates a collection of deal proposals for activation, and returns their combined weight,
+// split into regular deal weight and verified deal weight.
+func ValidateDealsForActivation(st *State, store adt.Store, dealIDs []abi.DealID, minerAddr addr.Address,
+	sectorExpiry, currEpoch abi.ChainEpoch) (big.Int, big.Int, error) {
+
+	proposals, err := AsDealProposalArray(store, st.Proposals)
+	if err != nil {
+		return big.Int{}, big.Int{}, fmt.Errorf("failed to load proposals: %w", err)
+	}
+
+	totalDealSpaceTime := big.Zero()
+	totalVerifiedSpaceTime := big.Zero()
+	for _, dealID := range dealIDs {
+		proposal, err := proposals.Get(dealID)
+		if err != nil {
+			return big.Int{}, big.Int{}, fmt.Errorf("failed to load deal %d: %w", dealID, err)
+		}
+		if err = validateDealCanActivate(proposal, minerAddr, sectorExpiry, currEpoch); err != nil {
+			return big.Int{}, big.Int{}, fmt.Errorf("cannot activate deal %d: %w", dealID, err)
+		}
+
+		// Compute deal weight
+		dealSpaceTime := DealWeight(proposal)
+		if proposal.VerifiedDeal {
+			totalVerifiedSpaceTime = big.Add(totalVerifiedSpaceTime, dealSpaceTime)
+		} else {
+			totalDealSpaceTime = big.Add(totalDealSpaceTime, dealSpaceTime)
+		}
+	}
+	return totalDealSpaceTime, totalVerifiedSpaceTime, nil
+}
+
 ////////////////////////////////////////////////////////////////////////////////
 // Checks
 ////////////////////////////////////////////////////////////////////////////////
 
-func validateDealCanActivate(rt Runtime, minerAddr addr.Address, sectorExpiration abi.ChainEpoch, proposal *DealProposal) {
+func validateDealCanActivate(proposal *DealProposal, minerAddr addr.Address, sectorExpiration abi.ChainEpoch, currEpoch abi.ChainEpoch) error {
```

Comment:
Nit: sectorExpiration and currEpoch could be combined into `sectorExpiration, currEpoch abi.ChainEpoch`.